libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -50,6 +50,7 @@ public class BotFrameworkAdapter : BotAdapter, IAdapterIntegration, IExtendedUse
 
         private static readonly HttpClient DefaultHttpClient = new HttpClient();
 
+        private readonly bool _isDefaultClient;
```

Comment:
>  private readonly bool _isDefaultClient; [](start = 6, length = 41)

I don't think you need this.  Just only set _httpClient when it's custom.

Reply:
_httpClient has always been set to something, either the customHttpClient or DefaultHttpClient.

I only wanted to impact the code to not use the DefaultHttpClient when one is not specified. Otherwise you do not and cannot use the retry policy that Rest implements. It is implemented by the handlers and you cannot adjust handlers after the fact unless there is some reference, which we lose when we provide the custom client.

_httpClient needs to be set to something otherwise we get into problems later on in a call to ConnectorClient.AddDefaultRequesetHeaders. It locks on _httpClient. We would need to change that as well if we let _httpClient be null.